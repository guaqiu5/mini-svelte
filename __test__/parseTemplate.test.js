const parseTemplate = require('../object/parseTemplate')

describe('parseTemplate', () => {
    it('解析html模板', () => {
        expect(parseTemplate(`<div class="wrapper"/>`)).toMatchSnapshot()
    })

    it('得到text', () => {
        expect(parseTemplate(`<div>Hello Svelte</div>`)).toMatchSnapshot()
    })

    it('不符合if each wait的block抛出异常', () => {
        expect(() => parseTemplate(`{#guaqiu }`)).toThrow('guaqiu is not support yet')
    })

    it(`ifblock结点`, () => {
        expect(parseTemplate(`{#if guaqiu}`)).toMatchSnapshot()
    })

    it('ifBlock缺失右边的括号报错', () => {
        expect(() => parseTemplate(`{#if hhhh`)).toThrow('block is missing')
    })

    it('eachBlock缺失as报错', () => {
        expect(() => parseTemplate(`{#each guaqiu}`)).toThrow('each Block needs as')
    })

    it('eachBlock结点(没有index)', () => {
        expect(parseTemplate(`{#each guaqius as item}`)).toMatchSnapshot()
    })

    it('eachBlock结点(有index)', () => {
        expect(parseTemplate(`{#each guaqius as item,index}`)).toMatchSnapshot()
    })

    it(`block的名字需要对应`, () => {
        expect(() => parseTemplate(`{#each guaqius as item,i}{/if}`)).toThrow('block name should match')
    })

    it('end block', () => {
        expect(parseTemplate(`{#each guaqius as item,i}{/each}`)).toMatchSnapshot()
    })
})