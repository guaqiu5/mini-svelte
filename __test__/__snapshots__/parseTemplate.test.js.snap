// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseTemplate eachBlock结点(有index) 1`] = `
Node {
  "attrs": [],
  "children": [
    Node {
      "attrs": [],
      "children": [],
      "data": {
        "index": "index",
        "localScopeVariable": "item",
        "variableExpression": "guaqius",
      },
      "end": 0,
      "name": "",
      "selfClosing": false,
      "start": 2,
      "type": "each Block",
    },
  ],
  "data": "",
  "end": 29,
  "name": "",
  "selfClosing": false,
  "start": 0,
  "type": "Fragment",
}
`;

exports[`parseTemplate eachBlock结点(没有index) 1`] = `
Node {
  "attrs": [],
  "children": [
    Node {
      "attrs": [],
      "children": [],
      "data": {
        "localScopeVariable": "item",
        "variableExpression": "guaqius",
      },
      "end": 0,
      "name": "",
      "selfClosing": false,
      "start": 2,
      "type": "each Block",
    },
  ],
  "data": "",
  "end": 23,
  "name": "",
  "selfClosing": false,
  "start": 0,
  "type": "Fragment",
}
`;

exports[`parseTemplate end block 1`] = `
Node {
  "attrs": [],
  "children": [
    Node {
      "attrs": [],
      "children": [],
      "data": {
        "index": "i",
        "localScopeVariable": "item",
        "variableExpression": "guaqius",
      },
      "end": 32,
      "name": "",
      "selfClosing": false,
      "start": 2,
      "type": "each Block",
    },
  ],
  "data": "",
  "end": 32,
  "name": "",
  "selfClosing": false,
  "start": 0,
  "type": "Fragment",
}
`;

exports[`parseTemplate ifblock结点 1`] = `
Node {
  "attrs": [],
  "children": [
    Node {
      "attrs": [],
      "children": [],
      "data": {
        "ifCondition": "guaqiu",
      },
      "end": 0,
      "name": "",
      "selfClosing": false,
      "start": 2,
      "type": "if Block",
    },
  ],
  "data": "",
  "end": 12,
  "name": "",
  "selfClosing": false,
  "start": 0,
  "type": "Fragment",
}
`;

exports[`parseTemplate 得到text 1`] = `
Node {
  "attrs": [],
  "children": [
    Node {
      "attrs": [],
      "children": [
        Node {
          "attrs": [],
          "children": [],
          "data": "Hello Svelte",
          "end": 17,
          "name": "",
          "selfClosing": false,
          "start": 5,
          "type": "Text",
        },
      ],
      "data": "",
      "end": 22,
      "name": "div",
      "selfClosing": false,
      "start": 0,
      "type": "Element",
    },
  ],
  "data": "",
  "end": 23,
  "name": "",
  "selfClosing": false,
  "start": 0,
  "type": "Fragment",
}
`;

exports[`parseTemplate 解析html模板 1`] = `
Node {
  "attrs": [],
  "children": [
    Node {
      "attrs": [
        {
          "name": "class",
          "type": "Attribute",
          "value": "wrapper",
        },
      ],
      "children": [],
      "data": "",
      "end": 22,
      "name": "div",
      "selfClosing": true,
      "start": 0,
      "type": "Element",
    },
  ],
  "data": "",
  "end": 22,
  "name": "",
  "selfClosing": false,
  "start": 0,
  "type": "Fragment",
}
`;
